### ISSUE TITLE
Presharding input in maxpool2d

### ISSUE THREAD

**Issue by jnie-TT** [tenstorrent/tt-mlir#855]
Currently we're presharding inputs in maxpool2d in ttnn runtime, ideally we would like to insert an op in the compiler on the mlir level that accomplishes this. Creating an issue to track.

**Comment by sdjordjevicTT** [comment]
Can we model this into MLIR instead of doing a workaround in the runtime?

**Comment by jnie-TT** [comment]
> Can we model this into MLIR instead of doing a workaround in the runtime?

Yeah that's what this issue is for, I'll update the comment as it might not be clear of the intent, it should say "ideally we want to insert an op on on the mlir level that accomplishes this" :))

**Comment by sdjordjevicTT** [comment]
Yeah, please update it to make it clear :) I will put this into the MLIR Dialect\Passes component.
